src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDecreaseInheritedMemberVisibility.Fixer.cs

```diff
@@ -1,36 +1,88 @@
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var expression = root.FindNode(context.Span);
+
+            if (expression != null)
+            {
+                var title = MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotDecreaseInheritedMemberVisibilityTitle;
+                context.RegisterCodeFix(CodeAction.Create(title, c => IncreaseVisibility(context.Document, expression, c), equivalenceKey: title), context.Diagnostics);
+            }
+        }
+
+        private async Task<Document> IncreaseVisibility(Document document, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var declaration = editor.Generator.GetDeclaration(node);
+            var symbol = editor.SemanticModel.GetDeclaredSymbol(node);
+            var ancestorTypes = symbol?.ContainingType?.GetBaseTypes() ?? Enumerable.Empty<INamedTypeSymbol>();
+            var hiddenOrOverriddenMembers = ancestorTypes.SelectMany(t => t.GetMembers(symbol?.Name));
+
+            // Check if a public member was overridden
+            if (hiddenOrOverriddenMembers.Any(s => s.DeclaredAccessibility == Accessibility.Public))
+            {
+                var property = (symbol as IMethodSymbol)?.AssociatedSymbol as IPropertySymbol;
+                if (property != null)
+                {
+                    // Can't explicitly set an accessor to public if the property is already public (accessor acceissbilities must be restricting)
+                    editor.SetAccessibility(declaration, Accessibility.NotApplicable);
+                }
+                else
+                {
+                    editor.SetAccessibility(declaration, Accessibility.Public);
+                }
+                return editor.GetChangedDocument();
+            }
+
+            // Otherwise, check if a protected member was overridden
+            if (hiddenOrOverriddenMembers.Any(s => s.DeclaredAccessibility == Accessibility.Protected))
+            {
+                var property = (symbol as IMethodSymbol).AssociatedSymbol as IPropertySymbol;
+                if (property?.DeclaredAccessibility == Accessibility.Protected)
+                {
+                    // Can't explicitly set an accessor to protected if the property is already protected (accessor acceissbilities must be restricting)
+                    editor.SetAccessibility(declaration, Accessibility.NotApplicable);
+                }
+                else
+                {
+                    editor.SetAccessibility(declaration, Accessibility.Protected);
+                }
+                return editor.GetChangedDocument();
+            }
+
```

Comment:
Don't you also need to do it for `Accessibility.ProtectedOrInternal`? And then you have the question of which one you do first. I mean, it's obvious that if there's both a `public` and a `protected` member in an ancestor, you want the derived member to be `public`. What if there's both a `protected` and a `protected internal` member in your list?

I guess the answer is you do `protected internal` first, because it has the greater visibility.